File: problems/2644-find-the-maximum-divisibility-score/solution.cpp
class Solution {
public:
    int maxDivScore(vector<int>& nums, vector<int>& divisors) {
        int maxScore = -1;
        int result = -1;
        for (int divisor : divisors) {
            int score = 0;
            for (int num : nums) {
                if (num % divisor == 0) {
                    score++;
                }
            }
            if (score > maxScore) {
                maxScore = score;
                result = divisor;
            } else if (score == maxScore && score != -1) {
                result = min(result, divisor);
            }
        }
        return result;
    }
};

File: problems/2644-find-the-maximum-divisibility-score/solution.py
class Solution:
    def maxDivScore(self, nums: List[int], divisors: List[int]) -> int:
        max_score = -1
        result = float('inf')
        for divisor in divisors:
            score = 0
            for num in nums:
                if num % divisor == 0:
                    score += 1
            if score > max_score:
                max_score = score
                result = divisor
            elif score == max_score:
                result = min(result, divisor)
        return result